crates/script/src/sequence.rs

```diff
@@ -80,51 +80,27 @@ impl Drop for ScriptSequenceKind {
     }
 }
 
-pub const DRY_RUN_DIR: &str = "dry-run";
-
 /// Helper that saves the transactions sequence and its state on which transactions have been
 /// broadcasted
 #[derive(Clone, Default, Serialize, Deserialize)]
-pub struct ScriptSequence {
-    pub transactions: VecDeque<TransactionWithMetadata>,
-    pub receipts: Vec<AnyTransactionReceipt>,
-    pub libraries: Vec<String>,
-    pub pending: Vec<TxHash>,
-    #[serde(skip)]
-    /// Contains paths to the sequence files
-    /// None if sequence should not be saved to disk (e.g. part of a multi-chain sequence)
-    pub paths: Option<(PathBuf, PathBuf)>,
-    pub returns: HashMap<String, NestedValue>,
-    pub timestamp: u64,
-    pub chain: u64,
-    pub commit: Option<String>,
+pub struct ScriptSequenceManager {
+    inner: ScriptSequence,
 }
 
-/// Sensitive values from the transactions in a script sequence
-#[derive(Clone, Default, Serialize, Deserialize)]
-pub struct SensitiveTransactionMetadata {
-    pub rpc: String,
-}
+impl ScriptSequenceManager {
```

Comment:
do we only need a separate type for `forge-verify` stuff? if so I think we could just make it into free function or a function of `BroadcastedState`